control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -563,22 +564,29 @@ impl ResourceShutdownOperations for OperationGuardArc<VolumeSpec> {
     ) -> Result<(), SvcError> {
         let shutdown_nexuses = registry
             .specs()
-            .get_volume_shutdown_nexuses(&request.uuid)
+            .get_volume_shutdown_nexuses(request.uuid())
             .await;
-
         let mut result = Ok(());
+        let filter_needed = request.registered_targets().is_some();
         for nexus_res in shutdown_nexuses {
             match nexus_res.operation_guard_wait().await {
                 Ok(mut guard) => {
+                    let uuid = nexus_res.lock().clone().uuid;
+                    let nexus = registry.get_nexus(&uuid).await?;
+                    if filter_needed
+                        && filter_targets(request.registered_targets().unwrap_or_default(), nexus)?
```

Comment:
nit: why not have `filter_targets` receive an `Option` by itself, then we can avoid having `filter_needed` and `unwrap_or_default` at the caller site?

Reply:
Incorporated